### PR TITLE
Fixes for sdk

### PR DESCRIPTION
**Pull request by weichweich**
## fixes KILTprotocol/ticket#367

- make private key field in attester public
- get attributes from presentation request (TS API)
- refactor verifier to not be a class
- add null type to supported credential types (go internal)

## How to test:

- test, integration test, work through tutorial

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by weichweich** [comment]
@wischli please review if you have time. 🙂

File: src/types/Verification.ts [hunk 1]
```diff
@@ -31,8 +31,16 @@ export class VerificationSession extends String {
  * The message result of [[requestPresentation]] which should be sent to the [[Claimer]] and used in [[buildPresentation]].
  */
 export class PresentationRequest extends String {
-  // @ts-ignore
-  private thisIsOnlyHereToPreventClassMixes: int
+  public getRequestedProperties(): string[] {
```
Comment (wischli) [inline src/types/Verification.ts:34]: A test is missing for this. Do you want to do that or should I?

File: src/types/Verification.ts [hunk 2]
```diff
@@ -47,6 +55,14 @@ export class CombinedVerificationSession extends String {
...
-  private thisIsOnlyHereToPreventClassMixes: int
+  public getRequestedProperties(): string[][] {
+    const parsed = JSON.parse(this.valueOf())
+    if ('partialPresentationRequests' in parsed) {
+      const { partialPresentationRequests } = parsed
+      return partialPresentationRequests.map(
+        (req: { requestedAttributes: any }) => req.requestedAttributes
+      )
+    }
+    throw new Error('Invalid request')
```
Comment (wischli) [inline src/types/Verification.ts:66]: Also not handled by tests yet. Same question as above.

File: src/types/Verification.ts (hunk 1 above)
Comment (wischli) [reply src/types/Verification.ts:34]: To be more precise: Lines 32-39 and 63.

Comment (weichweich) [reply src/types/Verification.ts:34]: i will add a test for that

**Pull request merged**